### PR TITLE
RHOAIENG-29275: Migrate auth validation from webhook to CEL validation

### PR DESCRIPTION
**Pull request by ugiordan**
## Description

This PR migrates auth validation from webhook-based validation to CEL (Common Expression Language) validation for improved performance and simplified architecture.

**Key Changes:**
- **Add CEL validation rules** to `AuthSpec` in `auth_types.go`:
  - `AdminGroups`: Cannot contain 'system:authenticated' or empty strings
  - `AllowedGroups`: Cannot contain empty strings
- **Generate CRD manifests** with embedded CEL validation rules
- **Update e2e tests** to validate CEL validation behavior instead of webhook validation
- **Remove entire webhook infrastructure** for auth validation:
  - Deleted `internal/webhook/auth/` directory (4 files removed)
  - Removed auth webhook registration from `webhook.go`
  - Cleaned up auth webhook configuration from manifests
- **CEL validation tests** now run independently of webhook test flags

This change improves performance by moving validation from runtime webhook calls to API server-native CEL validation, reducing latency and eliminating webhook dependency for auth validation.

**JIRA Link:** [RHOAIENG-29275](https://issues.redhat.com/browse/RHOAIENG-29275)

## How Has This Been Tested?

- **Unit Tests**: Updated and validated auth e2e tests (`authcontroller_test.go`) to verify CEL validation behavior
- **CRD Generation**: Verified that `make manifests` generates CRDs with proper CEL validation rules
- **E2E Testing**: Confirmed that auth validation tests pass with CEL validation instead of webhook validation
- **Regression Testing**: Ensured existing auth functionality remains unchanged while validation mechanism is improved
- **Webhook Removal**: Verified that removal of auth webhook code doesn't break other webhook functionality

## Screenshot or short clip

N/A - This is an internal validation mechanism change with no UI impact.

## Merge criteria

- [x] You have read the [contributors guide](https://github.com/opendatahub-io/opendatahub-operator/blob/incubation/CONTRIBUTING.md).
- [x] Commit messages are meaningful - have a clear and concise summary and detailed explanation of what was changed and why.
- [x] Pull Request contains a description of the solution, a link to the JIRA issue, and to any dependent or related Pull Request.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

## Summary by CodeRabbit

* **New Features**
  * Added Kubernetes CEL validation rules enforcing constraints on `adminGroups` and `allowedGroups` in Auth resources.
  * Introduced integration tests to verify CEL validation enforcement at creation and update stages.
  * Added unit tests for authentication controller actions, service handler, and default auth resource creation logic.

* **Bug Fixes**
  * Improved documentation and error messages for `adminGroups` and `allowedGroups` constraints.

* **Chores**
  * Removed the validating admission webhook and all related webhook registrations and tests.
  * Updated end-to-end tests to validate group constraints via CEL instead of webhooks.
  * Refactored authentication controller code by exporting key functions and structs.
  * Added comprehensive unit tests for authentication controller actions and service handler logic.

### CODE REVIEW

File: tests/e2e/authcontroller_test.go [hunk 1]
```diff
@@ -298,79 +280,112 @@ func (tc *AuthControllerTestCtx) ValidateWebhookBlocksInvalidGroupsOnCreation(t
...
+			adminGroups:   []string{"", "", "valid-admin-group"},
+			allowedGroups: []string{"valid-allowed-group"},
+		},
+		{
+			name:          "multiple empty strings in AllowedGroups",
+			invalidValue:  "empty string",
+			fieldName:     "AllowedGroups",
+			adminGroups:   []string{"valid-admin-group"},
+			allowedGroups: []string{"", "valid-allowed-group", ""},
+		},
```
Comment (zdtsw) [inline tests/e2e/authcontroller_test.go]: need test these two even if  only-one empty has passed the test?

File: api/services/v1alpha1/auth_types.go [hunk 2]
```diff
@@ -32,7 +32,11 @@ var _ common.PlatformObject = (*Auth)(nil)
 
 // AuthSpec defines the desired state of Auth
 type AuthSpec struct {
-	AdminGroups   []string `json:"adminGroups"`
+	// AdminGroups cannot contain 'system:authenticated' (security risk) or empty strings
+	// +kubebuilder:validation:XValidation:rule="self.all(group, group != 'system:authenticated' && group != '')",message="AdminGroups cannot contain 'system:authenticated' or empty strings"
+	AdminGroups []string `json:"adminGroups"`
```
Comment (zdtsw) [inline api/services/v1alpha1/auth_types.go:38]: will 
```
spec:
   adminGroups: [] 
```
 be allowed by this validation?
as it is empty  i guess it will pass,  but it is not what we want, right?

Comment (ugiordan) [reply api/services/v1alpha1/auth_types.go:38]: Good catch! Yes, empty arrays would pass the original validation. Fixed by adding:
`+kubebuilder:validation:XValidation:rule="size(self) > 0"`

File: tests/e2e/authcontroller_test.go (hunk 1 above)
Comment (ugiordan) [reply tests/e2e/authcontroller_test.go]: You're right - these are redundant. The CEL `self.all()` fails on ANY empty string, so testing multiple empty strings doesn't add value.

**Comment by ugiordan** [comment]
@coderabbitai full review

**Review by zdtsw — APPROVED** [review]

**Review by zdtsw — APPROVED** [review]

**Comment by ugiordan** [comment]
Manually cherry-picked https://github.com/opendatahub-io/opendatahub-operator/pull/2254.
